tests/test_date.py

```diff
@@ -118,163 +191,225 @@ def test_should_one_date_for_each_month(self, low, high, expected_results):
                                 datetime(2014, 12, 31, 23, 59, 59, 0)]),
     ])
     def test_periods(self, low, high, period, expected_results):
-        results = list(date.get_intersecting_periods(low, high, period=period))
-        self.assertEquals(expected_results, results)
+        self.given_period_low(low)
+        self.given_period_high(high)
+        self.given_period(period)
+        self.given_expected_results(expected_results)
+        self.when_intersecting_period_calculated()
+        self.then_should_one_date_for_each_date()
+
+    def test_should_reject_easily_mistaken_dateutil_arguments(self):
+        periods = [
```

Comment:
Let's use `param` for clarity: this would clearly state how each test would be called.